Agw tracing, timeouts, retries (#410)
kgateway-dev/kgateway.dev: content/docs/main/resiliency/timeouts/about.md

---
title: About timeouts
weight: 5
description:
---

> A timeout is the amount of time ([duration](https://protobuf.dev/reference/protobuf/google.protobuf/#duration)) that the gateway waits for replies from a backend service before the service is considered unavailable. This setting can be useful to avoid your apps from hanging or fail if no response is returned in a specific timeframe. With timeouts, calls either succeed or fail within a predicatble timeframe.

Reviewer: ```suggestion
A timeout is the amount of time ([duration](https://protobuf.dev/reference/protobuf/google.protobuf/#duration)) that the gateway waits for replies from a backend service before the service is considered unavailable. This setting can be useful to avoid your apps from hanging or to fail if no response is returned in a specific timeframe. With timeouts, calls either succeed or fail within a predictable timeframe.
```

> The time an app needs to process a request can vary a lot which is why applying the same timeout across services can cause a variety of issues. For example, a timeout that is too long can result in excessive latency from waiting for replies from failing services. On the other hand, a timeout that is too short can result in calls failing unnecessarily while waiting for an operation that needs responses from multiple services.

Reviewer: ```suggestion
The time an app needs to process a request can vary a lot. For this reason, applying the same timeout across services can cause a variety of issues. For example, a timeout that is too long can result in excessive latency from waiting for replies from failing services. On the other hand, a timeout that is too short can result in calls failing unnecessarily while waiting for an operation that needs responses from multiple services.
```

## Configuration options

> You can configure different types of timeouts by using a Kubernetes Gateway API-native configuration or a {{< reuse "docs/snippets/kgateway.md" >}}{{< reuse "docs/snippets/trafficpolicy.md" >}} as shown in the following table.

Reviewer: similar comment on adding timeouts to the agentgateway limitation section
